File: Data Structures & Algorithms/valid-sudoku/submission-0.cpp
class Solution {
public:
    bool isValidSudoku(vector<vector<char>>& board) {
        vector<unordered_set<int>> rows(board.size());
vector<unordered_set<int>> cols(board.size()); 
        unordered_set<int> bloc(board.size()); 
        vector<unordered_set<int>> blocks(board.size());

        for (int row = 0; row < board.size() ; row++){
            for (int col = 0; col < board[0].size(); col++ )
            { 
                char c = board[row][col];
                if (c == '.') continue;
                int value = c;

                 if (rows[row].find(value) != rows[row].end())
                    return false; 
                else 
                    rows[row].insert(value);

                 if (cols[col].find(value) != cols[col].end())
                    return false;
                else
                    cols[col].insert(value);
                
                int blocknumber1 = blocknumber(row,col,board.size());
                if(blocks[blocknumber1].find(board[row][col])!=blocks[blocknumber1].end())
                    return false; 
                else 
                    blocks[blocknumber1].insert(board[row][col]);
                
            }
           
        }   
         return true; 
    }

int blocknumber(int row, int col, int boardsize) {
    int blocksize = static_cast<int>(round(sqrt(boardsize)));
    int blockIndex = (row / blocksize) * blocksize + (col / blocksize);
    return blockIndex;
}
};

File: Data Structures & Algorithms/surrounded-regions/submission-0.py
class Solution:
    def solve(self, board: List[List[str]]) -> None:
        rows = len(board)
        cols = len(board[0])
        stack = deque()
        visited = []
        neighbors = [(-1,0),(1,0),(0,1),(0,-1)]
          

        for col in range (cols):
            if board[0][col] == 'O':
                board[0][col]='#'
                stack.append((0,col))
                
            if board[rows-1][col]=='O':
                board[rows-1][col]='#'
                stack.append((rows-1,col))
                
        for row in range (rows):
            if board[row][cols-1]=='O':
                board[row][cols-1]='#'
                stack.append((row,cols-1))
                
            if board[row][0]=='O':
                board[row][0]='#'
                stack.append((row,0))
                
                
        while stack:
            r,c = stack.pop()
            visited.append(board[r][c])
            for ar,ac in neighbors:
                nr , nc = ar+r,ac+c
                
                if( 0 <= nr < rows and 0 <= nc < cols and board[nr][nc]=='O'):
                    board[nr][nc]='#'
                    stack.append((nr,nc))
                    

        for col in range (cols):
            for row in range (rows): 
                if(board[row][col]=='O'):
                    board[row][col]='X'
                if(board[row][col]=='#'):
                    board[row][col]='O'


            

        